packages/puppeteer-core/src/bidi/Frame.ts

```diff
@@ -258,19 +260,33 @@ export class BidiFrame extends Frame {
     });
   }
 
+  #detached$() {
+    if (this.detached) {
+      return of(this as Frame);
+    }
+    return fromEmitterEvent(
+      this.page().trustedEmitter,
+      PageEvent.FrameDetached
```

Comment:
should this also wait for its own child frames?

Reply:
All the child frames should be destroyed at this point.